src/cargo_communication/request_actor.rs

```diff
@@ -48,76 +49,69 @@ fn get_request_status_code(result: &io::Result<ExitStatus>) -> StatusCode {
     }
 }
 
-impl<R> RequestActor<R>
+impl<R, C> RequestActor<R, C>
 where
     R: Request,
     R::Params: CreateCommand,
     R::Result: CargoResult,
+    C: CargoHandleTrait<CargoMessage>,
 {
     pub fn new(
         sender: Box<dyn Fn(Message) + Send>,
         req_id: RequestId,
         params: R::Params,
         root_path: &Path,
-    ) -> RequestActor<R> {
+        cargo_handle: C,
+        cancel_receiver: Receiver<Event>,
+    ) -> RequestActor<R, C> {
         RequestActor {
             sender,
-            cargo_handle: None,
+            cargo_handle: Some(cargo_handle),
```

Comment:
Czy po tych zmianach cargo_handle musi być nadal Option?

Reply:
Jesli nie byloby option to musielibysmy przechowywac w jakis inny sposob informacje czy wyslalismy juz notification o zcancelowanym tasku